independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/stereotypes/TransitiveStereotypeTest.java

```diff
@@ -60,13 +65,23 @@ public void test() {
 
             Set<Bean<?>> metadata = bm.getBeans(MyBean.class);
             assertEquals(1, metadata.size());
-            assertEquals(RequestScoped.class, metadata.iterator().next().getScope());
+            assertMetadata((InjectableBean<?>) metadata.iterator().next());
 
             container.requestContext().deactivate();
         }
     }
 
+    private void assertMetadata(InjectableBean<?> bean) {
```

Comment:
Since you mention it in the commit msg, do we actually test a hierarchy of stereotypes that's at least two levels deep?
Also, testing the circular stereotype reference case might be good safety latch for future code changes.